src/type_checker/checker.rs

```diff
@@ -52,24 +53,44 @@ struct ExprTyInfo {
 
     /// The type of the expression node.
     typ: TyKind,
+
+    /// This is needed to pass around the const value (values not represented by variables)
+    /// that is not stored in the global constants hashmap.
+    /// For example:
+    /// let xx = [1; 10]; // where 10 is a const value that should be passed around in the type checker
+    value: Option<String>,
```

Comment:
interesting

Reply:
also isn't that information contained in `TyKind::Array(_, size)`?

Reply:
Oh right, this addition of `value` might be unnecessary. We just need `TyKind::Array(_, size)`